src/matlab2tikz.m

```diff
@@ -1957,21 +1973,33 @@ function assertRegularAxes(handle)
     end
 end
 % ==============================================================================
-function [m2t,str] = plotLine2d(m2t, opts, errorBarOpts, data)
-    errorbarMode = (size(data,2) == 4); % is (optional) yDeviation given?
+function [m2t,str] = plotLine2d(m2t, opts, errorBarOpts, data, errDir)
+    errorbarModeX = strcmp(errDir, 'x'); % is (optional) xDeviation given?
+    errorbarModeY = strcmp(errDir, 'y'); % is (optional) yDeviation given?
+    errorbarModeXY = strcmp(errDir, 'xy'); % are (optional) both xDeviation and yDeviation given?
 
     errorBar = '';
-    if errorbarMode
+    if errorbarModeX || errorbarModeY || errorbarModeXY
         m2t      = needsPgfplotsVersion(m2t, [1,9]);
         errorBar = sprintf('plot [%s]\n', errorBarOpts);
     end
 
     % Convert to string array then cell to call sprintf once (and no loops).
     [m2t, table, tableOptions] = makeTable(m2t, repmat({''}, size(data,2)), data);
-    if errorbarMode
+    if errorbarModeX
+        tableOptions = opts_add(tableOptions, 'x error plus index', '2');
+        tableOptions = opts_add(tableOptions, 'x error minus index', '3');
+    end
+    if errorbarModeY
         tableOptions = opts_add(tableOptions, 'y error plus index', '2');
         tableOptions = opts_add(tableOptions, 'y error minus index', '3');
     end
+    if errorbarModeXY
+        tableOptions = opts_add(tableOptions, 'x error plus index', '2');
+        tableOptions = opts_add(tableOptions, 'x error minus index', '3');
+        tableOptions = opts_add(tableOptions, 'y error plus index', '4');
+        tableOptions = opts_add(tableOptions, 'y error minus index', '5');
+    end
```

Comment:
```suggestion
    switch errDir
        case 'x'
          tableOptions = opts_add(tableOptions, 'x error plus index', '2');
          tableOptions = opts_add(tableOptions, 'x error minus index', '3');
        case 'y'
          tableOptions = opts_add(tableOptions, 'y error plus index', '2');
          tableOptions = opts_add(tableOptions, 'y error minus index', '3');
        case 'xy'
          tableOptions = opts_add(tableOptions, 'x error plus index', '2');
          tableOptions = opts_add(tableOptions, 'x error minus index', '3');
          tableOptions = opts_add(tableOptions, 'y error plus index', '4');
          tableOptions = opts_add(tableOptions, 'y error minus index', '5');
    end
```

I would write it like above (note it's been a while since I've written MATLAB code, so the syntax might be somewhat off).

Reason for that is two-fold:
1. this kind of switch-case reduces the (cyclomatic) complexity of the code (in this switch-case there are 4 possible options: 'x', 'y', 'xy' (and the implicit "otherwise"). In the `if`s that you wrote, the code implies there are 8 options instead (one takes the if part or the implicit else), and that is repeated 3 times. 
2. using the `switch` can also remove the 3 lines doing the `strcmp`, hence making the code more local (i.e. easier to understand without looking a few lines past)

Reply:
Thanks a lot for the feedback! I incorporated your changes suggested. The TikZ file created is still the same, so all good with your syntax 😉